target/linux/mvebu/cortexa9/base-files/etc/uci-defaults/35_turris-omnia_uboot-env

```diff
@@ -0,0 +1,44 @@
+# This must be sourced after 30_uboot-envtools
+
+. /lib/functions.sh
+
+board=$(board_name)
+
+case "$board" in
+cznic,turris-omnia)
+	# Do nothing if this is not the old U-Boot
+	grep -q 'U-Boot 2015.10-rc2' /dev/mtd0 || exit 0
+	# Do nothing if we already have distro_bootcmd
+	fw_printenv distro_bootcmd >/dev/null 2>/dev/null && exit 0
+	# Set the complete environment, since U-Boot does not merge the default environment from its own image!
+	fw_setenv -s - <<-"EOF"
+		baudrate 115200
+		bootdelay 3
+		ethact neta2
+		fdt_high 0x10000000
+		initrd_high 0x10000000
+		bootargs earlyprintk console=ttyS0,115200 rootfstype=btrfs rootdelay=2 root=b301 rootflags=subvol=@,commit=5 rw
+		bootcmd i2c dev 1; i2c read 0x2a 0x9 1 0x00FFFFF0; setexpr.b rescue *0x00FFFFF0; if test $rescue -ge 1; then echo BOOT RESCUE; run rescueboot; else echo BOOT eMMC FS; run mmcboot; setenv bootargs; run distro_bootcmd; fi
+		rescueboot i2c mw 0x2a.1 0x3 0x1c 1; i2c mw 0x2a.1 0x4 0x1c 1; mw.l 0x01000000 0x00ff000c; i2c write 0x01000000 0x2a.1 0x5 4 -s; setenv bootargs "$bootargs omniarescue=$rescue"; sf probe; sf read 0x1000000 0x100000 0x700000; bootz 0x1000000
+		mmcboot btrload mmc 0 ${kernel_addr_r} boot/zImage @ && btrload mmc 0 ${fdt_addr_r} boot/dtb @ && setenv bootargs ${bootargs} cfg80211.freg=${regdomain} && bootz ${kernel_addr_r} - ${fdt_addr_r}
```

Comment:
Wouldn't it be better if You replaced `btrload` with `load`? That way users could format eMMC with whatever file system their U-Boot supports. Could be that the old U-Boot might not support this aliasing, that's why the question.

Reply:
Quote from the commit message:

> The first 9 lines are a copy of the default environment of the old U-Boot release - only modified, to run "distro_bootcmd", in case "mmcboot" fails to boot the factory OS.

So, `mmcboot` (9th of the first 9 lines) is only there to boot the factory OS (TurrisOS), which is and has always been on BTRFS. In all other cases, `mmcboot` will not be successful and fall through. Then, `distro_bootcmd` takes over, and searches for a boot script instead.

Honestly, I don't see the point in moving from `btrload` to `load` here. Especially, since the normal `load` won't allow to specify the subvolume...

Reply:
> Quote from the commit message:
> 
> > The first 9 lines are a copy of the default environment of the old U-Boot release - only modified, to run "distro_bootcmd", in case "mmcboot" fails to boot the factory OS.
> 
> So, `mmcboot` (9th of the first 9 lines) is only there to boot the factory OS (TurrisOS), which is and has always been on BTRFS. In all other cases, `mmcboot` will not be successful and fall through. Then, `distro_bootcmd` takes over, and searches for a boot script instead.

Indeed, there's no point if mmcboot fails and something better takes over (I hate web interfaces where I have to scroll vertically).

> 
> Honestly, I don't see the point in moving from `btrload` to `load` here. Especially, since the normal `load` won't allow to specify the subvolume...

ACK.